react/features/invite/components/add-people-dialog/native/AddPeopleDialog.js

```diff
@@ -0,0 +1,466 @@
+                        element => item[finderKey] !== element[finderKey])
+                });
+            } else {
+                // Item is not selected yet, need to add to the list.
+                this.setState({
+                    inviteItems: [
+                        ...inviteItems,
+                        item
+                    ].sort(objectComparator('name'))
+                });
+            }
+        };
+    }
+
+    _onTypeQuery: string => void
+
+    /**
+     * Handles the typing event of the text field on the dialog and performs the
+     * search.
+     *
+     * @param {string} query - The query that is typed in the field.
+     * @returns {void}
+     */
+    _onTypeQuery(query) {
+        clearTimeout(this.searchTimeout);
+        this.searchTimeout = setTimeout(() => {
+            this.setState({
+                searchInprogress: true
+            }, () => {
+                this._performSearch(query);
+            });
+        }, 500);
+    }
+
+    /**
+     * Performs the actual search.
+     *
+     * @param {string} query - The query to search for.
+     * @returns {void}
+     */
+    _performSearch(query) {
+        this._query(query).then(results => {
+            const { inviteItems } = this.state;
+
+            let selectableItems = results.filter(result => {
+                switch (result.type) {
+                case 'phone':
+                    return result.number && !inviteItems.find(
+                        finderByKey('number', result.number));
+                case 'user':
+                    return result.user_id && !inviteItems.find(
+                        finderByKey('user_id', result.user_id));
+                default:
+                    return false;
+                }
+            });
+
+            selectableItems.sort(objectComparator('name'));
+
+            selectableItems = this.state.inviteItems.concat(selectableItems);
+
+            this.setState({
+                selectableItems
+            });
+        })
+        .finally(() => {
+            this.setState({
+                searchInprogress: false
+            }, () => {
+                this.inputFieldRef && this.inputFieldRef.focus();
+            });
+        });
+    }
+
+    _query: (string) => Promise<Array<Object>>;
+
+    _renderItem: Object => ?React$Element<*>
+
+    /**
+     * Renders a single item in the {@code FlatList}.
+     *
+     * @param {Object} flatListItem - An item of the data array of the
+     * {@code FlatList}.
+     * @param {number} index - The index of the currently rendered item.
+     * @returns {?React$Element<*>}
+     */
+    _renderItem(flatListItem, index) {
+        const { item } = flatListItem;
+        const { inviteItems } = this.state;
+        let selected = false;
+        let renderableItem;
+
+        switch (item.type) {
+        case 'phone':
+            selected = inviteItems.find(finderByKey('number', item.number));
```

Comment:
finderbyKey can be replaced by lodash's matchesProperty: https://lodash.com/docs/4.17.11#matchesProperty